File: scm-csharp/scheme/primitives/PrimitiveMakeDict.cs
namespace scheme;

public class PrimitiveMakeDict : Primitive
{
    public override string Name()
    {
        return "make-dict";
    }

    public override string Info()
    {
        return
            "Syntax: (make-dict)\n" +
            "Library: (scm core)\n" +
            "Description: Returns a new empty mutable dictionary (hash map) with string or symbol keys.\n" +
            "Example:\n" +
            "  (let ((d (make-dict)))\n" +
            "    (dict-put d \"key\" 42)\n" +
            "    (dict-get d \"key\")) => 42";
    }
    
    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 0, 0);
        return new Dictionary<string, object>();
    }
}

File: scm-java/src/main/scheme/primitives/PrimitiveMakeDict.java
package scheme.primitives;

import scheme.*;

import java.util.LinkedHashMap;

public class PrimitiveMakeDict extends Primitive {
    @Override
    public String name() {
        return "make-dict";
    }

    @Override
    public String info() {
        return "Syntax: (make-dict)\n" +
               "Library: (scm core)\n" +
               "Description: Returns a new empty mutable dictionary (hash map) with string or symbol keys.\n" +
               "Example:\n" +
               "  (let ((d (make-dict)))\n" +
               "    (dict-put d \"key\" 42)\n" +
               "    (dict-get d \"key\")) => 42";
    }
    
    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 0, 0);
        return new LinkedHashMap<String, Object>();
    }
}
